src/utils/__tests__/httpSignature/ec.test.ts

```diff
@@ -0,0 +1,210 @@
+      TE.fold(
+        () => T.of(""),
+        result => T.of(result)
+      )
+    )();
+    expect(pemKey).not.toBe(ecPublicKeyPem);
+  });
+});
+
+describe("Test FCI custom content TOS and Sign Challenges", () => {
+  TEST_CONTENT.forEach(content => {
+    it(`Test FCI custom content to sign for "${content.header}" to be computed the right way`, async () => {
+      const customContentSignatureBase = getCustomContentSignatureBase(
+        SIGNATURE_INPUT,
+        content.challenge,
+        content.header
+      );
+      expect(customContentSignatureBase!.signatureBase).toBe(
+        content.signatureBase
+      );
+    });
+  });
+
+  TEST_CONTENT.forEach(content => {
+    it(`Test FCI custom content signature for "${content.header}" to be verified the right way`, async () => {
+      const customContentSignatureBase = getCustomContentSignatureBase(
+        SIGNATURE_INPUT,
+        content.challenge,
+        content.header
+      );
+
+      const customContentChallenge = getCustomContentChallenge(
+        customContentSignatureBase!.signatureLabel!,
+        SIGNATURE
+      );
+
+      const result = await verifyCustomContentChallenge(
+        customContentSignatureBase!.signatureBase,
+        customContentChallenge!,
+        ecPublicKeyJwk
+      )();
+
+      expect(result).toBeTruthy();
+    });
+  });
+});
+
+describe("Test provided signature algorithms", () => {
+  TEST_CONTENT.forEach(content => {
+    it(`Test that we retrive a valid sign algorithm for "${content.header}" from signature-input`, () => {
+      const signAlgorithm = getSignatureInfo(content.signatureBase);
+      expect(O.isSome(signAlgorithm)).toBeTruthy();
+      expect(isSignAlgorithmValid(signAlgorithm)).toBeTruthy();
+    });
+    it(`Test that we retrive a wrong sign algorithm for "${content.header}" from signature-input`, () => {
+      const wrongSignAlgorithm = getSignatureInfo("wrong-value");
+      expect(O.isNone(wrongSignAlgorithm)).toBeTruthy();
+      expect(isSignAlgorithmValid(wrongSignAlgorithm)).toBeFalsy();
+    });
+  });
+});
+
+describe("Test http-signature request", () => {
+  ["sig1", "sig2", "sig3", undefined, "sig8"].forEach(sigLabel => {
+    const mockRequestOptions: VerifySignatureHeaderOptions = {
+      verifier: {
+        verify: signAlgorithmToVerifierMap["ecdsa-p256-sha256"].verify(
+          ecPublicKeyJwk
+        )
+      },
+      url: "http://127.0.0.1:3000",
+      method: "GET",
+      httpHeaders: {
+        host: "127.0.0.1:3000",
+        connection: "Keep-Alive",
+        "accept-encoding": "gzip",
+        "user-agent": "okhttp/4.9.2",
+        "x-pagopa-lollipop-original-url": "/api/v1/profile",
+        "x-pagopa-lollipop-original-method": "GET",
+        [TOS_HEADER_NAME]: TOS_CHALLENGE,
+        [SIGNATURE_HEADER_NAME]: CHALLENGE,
+        signature: SIGNATURE,
+        "signature-input": SIGNATURE_INPUT
+      },
+      body: undefined,
+      signatureKey: sigLabel,
+      verifyExpiry: false
+    };
```

Comment:
What do you think about move this out of there at the top of file and write instead 
```suggestion
  ["sig1", "sig2", "sig3", undefined, "sig8"].forEach(signatureKey => {
    const mockRequestOptions: VerifySignatureHeaderOptions = {
      ...mockedRequest,
      signatureKey
    };
```

Reply:
I agree.